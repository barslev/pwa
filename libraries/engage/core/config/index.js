export { default as subscriptions } from './config.subscriptions';
export { default as reducers } from './config.reducers';

export * from './config.actions';
export * from './config.selectors';
export * from './config.streams';
