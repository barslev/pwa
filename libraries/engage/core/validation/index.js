export { validate } from './validation';
export { useValidation } from './validation.hooks';
