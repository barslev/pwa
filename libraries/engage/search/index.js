/** @module search */

// ACTIONS
export { default as fetchSearchResults } from '@shopgate/pwa-common-commerce/search/actions/fetchSearchResults';
export { default as fetchSearchSuggestions } from '@shopgate/pwa-common-commerce/search/actions/fetchSearchSuggestions';

// CONSTANTS
export * from '@shopgate/pwa-common-commerce/search/constants/index';
export * from '@shopgate/pwa-common-commerce/search/constants/Pipelines';
export * from '@shopgate/pwa-common-commerce/search/constants/Portals';

// HELPERS
export { default as removeHighlightingPlaceholers } from '@shopgate/pwa-common-commerce/search/helpers/removeHighlightingPlaceholders';
export * from '@shopgate/pwa-common-commerce/search/helpers';

// SELECTORS
export * from '@shopgate/pwa-common-commerce/search/selectors';

// STREAMS
export * from '@shopgate/pwa-common-commerce/search/streams';
