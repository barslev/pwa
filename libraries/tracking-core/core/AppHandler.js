import { SGAction } from '../helpers/helper';

/**
 * Handler for the communication between the tracking plugins and the app.
 */
class AppHandler {
  /**
   * Log a pageview
   *
   * @param {UnifiedPageview} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  viewContent(data, restrictions) {
    SGAction.analyticsLogPageview(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * AnalyticsLogEvent
   * @param {UnifiedPageview} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  logEvent(data, restrictions) {
    SGAction.analyticsLogEvent(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log a purchase
   *
   * @param {UnifiedPurchase} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  purchase(data, restrictions) {
    SGAction.analyticsLogPurchase(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a product was added to the cart
   *
   * @param {UnifiedAddToCart} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  addToCart(data, restrictions) {
    SGAction.analyticsLogAddToCart(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a payment info was added
   *
   * @param {UnifiedPaymentInfo} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  addedPaymentInfo(data, restrictions) {
    SGAction.analyticsLogAddedPaymentInfo(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a payment info was selected
   *
   * @param {UnifiedPaymentInfo} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  selectedPaymentInfo(data, restrictions) {
    SGAction.analyticsLogAddedPaymentInfo(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a checkout is initiated
   *
   * @param {UnifiedInitiatedCheckout} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  initiatedCheckout(data, restrictions) {
    SGAction.analyticsLogInitiatedCheckout(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log if a registration was completed
   *
   * @param {UnifiedCompletedRegistration} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  completedRegistration(data, restrictions) {
    SGAction.analyticsLogCompletedRegistration(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a product was added to the wishlist
   *
   * @param {UnifiedAddToWishlist} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  addToWishlist(data, restrictions) {
    SGAction.analyticsLogAddToWishlist(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when a search happened
   *
   * @param {UnifiedSearched} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  search(data, restrictions) {
    SGAction.analyticsLogSearch(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log when Deeplink and etc opened
   * @param {Object} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  setCampaignWithUrl(data, restrictions) {
    SGAction.analyticsSetCampaignWithUrl(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log an itemview
   *
   * @param {UnifiedItemView} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  itemView(data, restrictions) {
    SGAction.analyticsLogItemView(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Log a login
   *
   * @param {UnifiedLogin} data Tracking data for this event
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {AppHandler} Instance of SgTrackingAppHandler
   */
  login(data, restrictions) {
    SGAction.analyticsLogLogin(AppHandler.prepareTrackingData(data, restrictions));
    return this;
  }

  /**
   * Add the properties restriction, blacklist and trackers to the tracking event data if needed
   *
   * @param {Object} data The tracking event data
   * @param {UnifiedRestrictions} [restrictions] Info about the restrictions
   * @returns {Object} Tracking data with restrictions
   * @private
   */
  static prepareTrackingData(data, restrictions) {
    // Clone the data to avoid issues with references.
    const clonedData = { ...data };

    if (restrictions && restrictions.trackers) {
      let trackerParamName = 'trackers';

      // There is a Bug in the iOS app. Lib Version <= 9.0 expects "tracker" as the param name.
      if (
        window.LibShopgateClient &&
        window.LibShopgateClient.isIos() &&
        window.LibShopgateVersion &&
        window.LibShopgateVersion.isLibVersionAtMost(9)
      ) {
        trackerParamName = 'tracker';
      }

      // Add the restrictions to the tracking event data.
      clonedData.restrictions = {
        blacklist: restrictions.blacklist,
        [trackerParamName]: restrictions.trackers,
      };
    }

    return clonedData;
  }
}

export default new AppHandler();
