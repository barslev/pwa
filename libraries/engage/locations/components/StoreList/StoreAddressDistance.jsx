// @flow
import React from 'react';
import { I18n } from '@shopgate/engage/components';
import { i18n } from '@shopgate/engage/core';
import { storeDistance } from './style';

type Props = {
  distance?: number;
  unitSystem?: string;
}

export const UNIT_SYSTEM_METRIC = 'metric';
export const UNIT_SYSTEM_IMPERIAL = 'imperial';

/**
 * Renders a single store distance.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const StoreAddressDistance = ({ distance, unitSystem = UNIT_SYSTEM_METRIC }: Props) => {
  if (distance === null) {
    return null;
  }

  return (
    <I18n.Text
      className={storeDistance.toString()}
      string={`locations.distance_${unitSystem}`}
      params={{ distance: i18n.number(distance, 2) }}
    />
  );
};

StoreAddressDistance.defaultProps = {
  distance: null,
  unitSystem: UNIT_SYSTEM_METRIC,
};

export default StoreAddressDistance;
