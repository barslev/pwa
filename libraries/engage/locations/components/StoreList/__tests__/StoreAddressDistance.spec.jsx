import React from 'react';
import { shallow } from 'enzyme';
import { i18n } from '@shopgate/engage/core';
import StoreAddressDistance, { UNIT_SYSTEM_METRIC, UNIT_SYSTEM_IMPERIAL } from '../StoreAddressDistance';

jest.mock('@shopgate/engage/components', () => ({
  I18n: {
    Text: function Translate() { return null; },
  },
}));

jest.mock('@shopgate/engage/core', () => ({
  i18n: {
    number: jest.fn().mockImplementation(number => `${number}`),
  },
}));

describe('<StoreAddressDistance', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should not render when no distance was passed', () => {
    const wrapper = shallow((<StoreAddressDistance />));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toEqual(true);
  });

  it('should render when the distance is 0', () => {
    const distance = 0;
    const wrapper = shallow((<StoreAddressDistance distance={distance} />));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Translate').prop('string')).toEqual(`locations.distance_${UNIT_SYSTEM_METRIC}`);
    expect(wrapper.find('Translate').prop('params')).toEqual({ distance: '0' });
    expect(i18n.number).toHaveBeenCalledTimes(1);
    expect(i18n.number).toHaveBeenCalledWith(distance, 2);
  });

  it('should render when a unit system is set', () => {
    const distance = 5.4;
    const wrapper = shallow((
      <StoreAddressDistance
        distance={distance}
        unitSystem={UNIT_SYSTEM_IMPERIAL}
      />
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Translate').prop('string')).toEqual(`locations.distance_${UNIT_SYSTEM_IMPERIAL}`);
    expect(wrapper.find('Translate').prop('params')).toEqual({ distance: '5.4' });
    expect(i18n.number).toHaveBeenCalledTimes(1);
    expect(i18n.number).toHaveBeenCalledWith(distance, 2);
  });
});
