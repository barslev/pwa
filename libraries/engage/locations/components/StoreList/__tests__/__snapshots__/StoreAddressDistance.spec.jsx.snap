// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreAddressDistance should not render when no distance was passed 1`] = `""`;

exports[`<StoreAddressDistance should render when a unit system is set 1`] = `
<Translate
  className="css-1fxtar0"
  params={
    Object {
      "distance": "5.4",
    }
  }
  string="locations.distance_imperial"
/>
`;

exports[`<StoreAddressDistance should render when the distance is 0 1`] = `
<Translate
  className="css-1fxtar0"
  params={
    Object {
      "distance": "0",
    }
  }
  string="locations.distance_metric"
/>
`;
